redbot/cogs/audio/audio.py

```diff
@@ -3727,89 +3727,92 @@ def _restart_connect(self):
-
-        try:
-            playlist_id, playlist_arg = await self._get_correct_playlist_id(
-                ctx, playlist_matches, scope, author, guild, specified_user
-            )
-        except TooManyMatches as e:
-            return await self._embed_msg(ctx, str(e))
-        if playlist_id is None:
-            return await self._embed_msg(
-                ctx, _("Could not match '{arg}' to a playlist.").format(arg=playlist_arg)
+        async with ctx.typing():
+            if scope_data is None:
+                scope_data = [PlaylistScope.GUILD.value, ctx.author, ctx.guild, False]
+            scope, author, guild, specified_user = scope_data
+            scope_name = humanize_scope(
+                scope, ctx=guild if scope == PlaylistScope.GUILD.value else author
             )
 
-        try:
-            playlist = await get_playlist(playlist_id, scope, self.bot, guild, author)
-        except RuntimeError:
-            return await self._embed_msg(
-                ctx,
-                _("Playlist {id} does not exist in {scope} scope.").format(
-                    id=playlist_id, scope=humanize_scope(scope, the=True)
-                ),
-            )
-        except MissingGuild:
-            return await self._embed_msg(
-                ctx, _("You need to specify the Guild ID for the guild to lookup.")
-            )
+            try:
+                playlist_id, playlist_arg = await self._get_correct_playlist_id(
+                    ctx, playlist_matches, scope, author, guild, specified_user
+                )
+            except TooManyMatches as e:
+                return await self._embed_msg(ctx, str(e))
+            if playlist_id is None:
+                return await self._embed_msg(
+                    ctx, _("Could not match '{arg}' to a playlist.").format(arg=playlist_arg)
+                )
 
-        if not await self.can_manage_playlist(scope, playlist, ctx, author, guild):
-            return
+            try:
+                playlist = await get_playlist(playlist_id, scope, self.bot, guild, author)
+            except RuntimeError:
+                return await self._embed_msg(
+                    ctx,
+                    _("Playlist {id} does not exist in {scope} scope.").format(
+                        id=playlist_id, scope=humanize_scope(scope, the=True)
+                    ),
+                )
+            except MissingGuild:
+                return await self._embed_msg(
+                    ctx, _("You need to specify the Guild ID for the guild to lookup.")
+                )
 
-        track_objects = playlist.tracks_obj
-        original_count = len(track_objects)
-        unique_tracks = set()
-        unique_tracks_add = unique_tracks.add
-        track_objects = [
-            x for x in track_objects if not (x in unique_tracks or unique_tracks_add(x))
-        ]
+            if not await self.can_manage_playlist(scope, playlist, ctx, author, guild):
+                return
```

Comment:
Looks good to me.
Although imo the context manager should be added after this line.
All the heavy work is done after it so thats where it would be relevant, Also showing typing for a split second for users that don't have the perms would go against all the other perm checks in audio. but i'll let @aikaterna  have the final say in regards to the UX here

Reply:
Meh ignore me me here, this will make it clearer to the user  if typing starts at the very start of the command, specialy once there several playlists added